kebbie/emulator.py

```diff
@@ -345,37 +347,38 @@ def __init__(  # noqa: C901
-        elif self.keyboard == FLEKSY:
-            self.detected = FleksyLayoutDetector(self.driver)
-            self.layout = self.detected.layout
-        elif self.keyboard == IOS:
-            self.detected = IosLayoutDetector(self.driver, self._tap)
-            self.layout = self.detected.layout
-        elif self.keyboard == KBKITPRO:
-            self.detected = KbkitproLayoutDetector(self.driver, self._tap)
-            self.layout = self.detected.layout
-        elif self.keyboard == KBKITOSS:
-            self.detected = KbkitossLayoutDetector(self.driver, self._tap)
-            self.layout = self.detected.layout
-        elif self.keyboard == SWIFTKEY:
-            self.detected = SwiftkeyLayoutDetector(self.driver, self._tap)
-            self.layout = self.detected.layout
-        elif self.keyboard == YANDEX:
-            self.detected = YandexLayoutDetector(self.driver, self._tap)
-            self.layout = self.detected.layout
-        else:
-            raise ValueError(
-                f"Unknown keyboard : {self.keyboard}. Please specify `{GBOARD}`, `{TAPPA}`, `{FLEKSY}`, "
-                f"`{SWIFTKEY}`, `{YANDEX}`, `{KBKITPRO}`, `{KBKITOSS}` or `{IOS}`."
-            )
+        if get_layout:
+            if self.keyboard == GBOARD:
+                self.detected = GboardLayoutDetector(self.driver, self._tap)
+                self.layout = self.detected.layout
+            elif self.keyboard == TAPPA:
+                self.detected = TappaLayoutDetector(self.driver, self._tap)
+                self.layout = self.detected.layout
+            elif self.keyboard == FLEKSY:
+                self.detected = FleksyLayoutDetector(self.driver)
+                self.layout = self.detected.layout
+            elif self.keyboard == IOS:
+                self.detected = IosLayoutDetector(self.driver, self._tap)
+                self.layout = self.detected.layout
+            elif self.keyboard == KBKITPRO:
+                self.detected = KbkitproLayoutDetector(self.driver, self._tap)
+                self.layout = self.detected.layout
+            elif self.keyboard == KBKITOSS:
+                self.detected = KbkitossLayoutDetector(self.driver, self._tap)
+                self.layout = self.detected.layout
+            elif self.keyboard == SWIFTKEY:
+                self.detected = SwiftkeyLayoutDetector(self.driver, self._tap)
+                self.layout = self.detected.layout
+            elif self.keyboard == YANDEX:
+                self.detected = YandexLayoutDetector(self.driver, self._tap)
+                self.layout = self.detected.layout
+            else:
+                raise ValueError(
+                    f"Unknown keyboard : {self.keyboard}. Please specify `{GBOARD}`, `{TAPPA}`, `{FLEKSY}`, "
+                    f"`{SWIFTKEY}`, `{YANDEX}`, `{KBKITPRO}`, `{KBKITOSS}` or `{IOS}`."
+                )
 
-        self.typing_field.clear()
+            self.typing_field.clear()
```

Comment:
```suggestion
        self.typing_field.clear()
```

We should clear the typing field even though we don't get the layout